httpql/src/main/java/com/hubspot/httpql/internal/CombinedFilterEntry.java

```diff
@@ -0,0 +1,107 @@
+package com.hubspot.httpql.internal;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jooq.Condition;
+import org.jooq.Operator;
+import org.jooq.impl.DSL;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.hubspot.httpql.FieldFactory;
+import com.hubspot.httpql.QuerySpec;
+
+public class CombinedFilterEntry<T extends QuerySpec> implements FilterEntryConditionCreator<T> {
+
+  private Operator operator;
+
+  private List<FilterEntryConditionCreator<T>> conditionCreators;
+
+  public CombinedFilterEntry(Operator operator, List<FilterEntryConditionCreator<T>> conditionCreators) {
+    this.operator = operator;
+    this.conditionCreators = Lists.newArrayList(conditionCreators);
+  }
+
+  @Override
+  public Condition getCondition(QuerySpec value, FieldFactory fieldFactory) {
+    return DSL.condition(operator, conditionCreators.stream()
+        .map(cc -> cc.getCondition(value, fieldFactory))
+        .collect(Collectors.toList()));
+  }
+
+  @Override
+  public Collection<BoundFilterEntry<T>> getFlattenedBoundFilterEntries() {
+    return conditionCreators.stream()
+        .map(FilterEntryConditionCreator::getFlattenedBoundFilterEntries)
+        .flatMap(Collection::stream)
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  public void addConditionCreator(FilterEntryConditionCreator<T> conditionCreator) {
+    if (!getFlattenedBoundFilterEntries().contains(conditionCreator)) {
```

Comment:
Would it be worth memoizing the result of `getFlattenedBoundFilterEntries()` (and resetting on each addition to `conditionCreators`)?

Reply:
I thought about this but since `conditionCreators` is an exposed mutable list, it'd be especially difficult to determine when the memoized value went stale.